stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -425,12 +460,134 @@ extension UnsafeMutableBufferPointer {
   ///   buffer, and an index to the point in the buffer one past the last
   ///   element written.
   @_inlineable
-  public func initialize<S: Sequence>(from source: S) -> (S.Iterator, Index)
-    where S.Element == Element {
+  @discardableResult
+  public func initialize<S>(from source: S) -> (S.Iterator, Index)
+    where S:Sequence, S.Element == Element {
     return source._copyContents(initializing: self)
   }
+
+  // Warning: All buffer elements must be initialized before calling this. Note 
+  // that invoking initialize(from:) may not initialize all elements. 
+  // Assigning to part of the buffer must be done using the assign(repeating:count:) 
+  // method on the buffer’s baseAddress. 
+  @_inlineable
+  public func assign(repeating repeatedValue: Element) {
+    guard let dstStart = _position else {
+      return
+    }
+
+    dstStart.assign(repeating: repeatedValue, count: _end! - dstStart)
+  }
+
+  // Warning: All buffer elements must be initialized before calling this. Note 
+  // that invoking initialize(from:) may not initialize all elements. 
+  // Assigning to part of the buffer must be done using the assign(from:count:) 
+  // method on the buffer’s baseAddress. 
+  @_inlineable
+  @discardableResult
+  public func assign<S>(from source: S) -> (S.Iterator, Index)
+    where S:Sequence, S.Element == Element {
+    return source._copyContents(assigningTo: self)
+  }
+
+  // Warning: All buffer elements must be initialized before calling this. Note 
+  // that invoking initialize(from:) may not initialize all elements. 
+  // Assigning to part of the buffer must be done using the moveAssign(from:count:) 
+  // method on the buffer’s baseAddress. 
+  @_inlineable
+  public func moveAssign(from source: UnsafeMutableBufferPointer<Element>) {
+    _debugPrecondition(source.count >= count,
+                      "${Self}.moveAssign(from:) source does not have enough elements")
+    guard let dstStart = _position,
+          let srcStart = source._position else {
```

Comment:
is there a stdlib convention for this? my own convention is to always put `guard` and `if` comma statements on their own lines